Create 1-introduction.md (#31)
Vaibhavpratapsingh22/Courses: 4-software-engineering/README.md

# Software Engineering

Software Engineering is the field of computer science that deals with the construction of large or complex software system. The software development of this magnitude is usually done by a group of software professionals rather than the single professional. The differnce between programming and software engineering is that programming is primiarly activity while software engineering is essentially a team activity.

## Chapter 1 - Introduction

It is about the basics of software engineering. We will start with how a program is different from software. Then we will head towards the software crisis. In the software crisis, we will learn its causes, characteristics. At last, we will learn about the elements of software engineering.

> ## Chapter 2 -  Software Development Process Models

Reviewer: from my point of view - Chapter 2 shouldn't fall on this branch already. 
How would we deal with the situation like we are in now - i have suggested some changes on other branch regarding chapter 2 abstract there. 
There will be a merge conflict for sure. how would we deal with it in that case? will we take copy of README from master or `se-development-models`? 
Put some time on learning a bit more about clean branching work. 
@giteshk11 we are facing some troubles here regarding branch creation. Could you help here?

Author: I think @udeysingh95  now both the branches are in perfect state. But @giteshk11 if u know anything about how to tackle with different branches without any conflict plz guide me.

Reviewer 2: Hi @Vaibhavpratapsingh22 what issue are you facing?

Author: Hlo @giteshk11 basically I have made two branches but when i do any commit then confilct  occurs. Although i have rebased se-development-models on se-introduction.

Reviewer 2: You should pull the changes from the branch first and then merge it then commit it

# Software Engineering

Software Engineering is the field of computer science that deals with the construction of large or complex software system. The software development of this magnitude is usually done by a group of software professionals rather than the single professional. The differnce between programming and software engineering is that programming is primiarly activity while software engineering is essentially a team activity.

## Chapter 1 - Introduction

> In this chapter we will look at the basics of software engineering like :

Reviewer: ```suggestion
It is about the basics of software engineering. We will start with how a program is different from software. Then we will head towards the software crisis. In the software crisis, we will learn its causes, characteristics. At last, we will learn about the elements of software engineering.
```

> 1. Definition of Software

Reviewer: I would not just write the section names here. I would suggest to write a short paragraph here - which can say everything about chapter 1. but not in points or not the way its given on issue. But in your words - like a short abstract. If you are not familiar with **Abstract**. you can read about how to write abstracts and then write one for each chapter here. 3-4 lines is good size here. It would keep our READMe short as well and faster to understand the whole work.

Author: okay working on it .

> 1. Definition of Software

Reviewer: ```suggestion
```

>    * Program versus Software
>    * Software Crisis
> 2. Software Characteristics
> 3. Definition of Software Engineering
> 4. Elements of Software Engineering

Reviewer: ```suggestion
```

> # Software Engineering

Reviewer: This file has been directly taken from the issue explanation. It doesn't need to be same. you can explain it in your words. You can keep the chapter names as it is but leave the sub-points and put summary in your own words. This could be something similar to the chapter summary that i mentioned in one of my previous comments.

Software Engineering is the field of computer science that deals with the construction of large or complex software system. The software development of this magnitude is usually done by a group of software professionals rather than the single professional. The differnce between programming and software engineering is that programming is primiarly activity while software engineering is essentially a team activity.

You will gain not only knowledge and practical experience of the latest technologies, but also a grounding in the underlying principles of the subject. It is this combination of skills that will us to keep pace with this fast-evolving subject!

## Chapter 1 - Introduction

In this chapter we will look at the basics of software engineering like :

1. Definition of Software
   * Program versus Software
   * Software Crisis
2. Software Characteristics
3. Definition of Software Engineering
4. Elements of Software Engineering

> 

Reviewer: Also, this README actually only contains the content that this folder holds - this shouldn't be the case. This readme should be already in state to provide some overview to whats coming in here. 
so you can actually move at least the chapter names here and add some closing words in readme. 
and then ofcourse we will evolve it with time and PRs.

Author: If I only metioned the sub headings of the topic it may get very long at the end and it may appear like the main README file .

Reviewer: Good point - thats why we need to find a better way to write these README files
